Refactor/166/comment section

rak517 opened the pull request:
## ❓이슈
- close #166 

## :writing_hand: Description

- 댓글 수정 시 변경사항이 없으면 submit 불가능하게 변경
- 댓글 리스트 영역에 스크롤 부여해서 Card의 길이가 변경되지 않도록 수정
- 로딩 시 스켈레톤 UI 추가
![스크린샷 2025-02-17 14 51 42](https://github.com/user-attachments/assets/5dd8fb20-4a1d-4aaa-b0ed-70ff7bf1ae84)
![스크린샷 2025-02-17 14 51 56](https://github.com/user-attachments/assets/006615c1-aa1d-40fa-819c-b5e0d3634896)
![스크린샷 2025-02-17 15 52 51](https://github.com/user-attachments/assets/d637d920-84d4-427a-825c-2c0e361c7afa)

## :white_check_mark: Checklist

### PR

<!-- 작성중인 PR인 경우, Draft 모드로 생성해주세요. -->

- [x] Branch Convention 확인
  > `feat/*` 기능 구현, `fix/*` 버그 수정, `refactor/*` 개선
- [x] Base Branch 확인
- [x] 적절한 Label 지정
- [x] Assignee 및 Reviewer 지정

### Test

- [x] 로컬 작동 확인

### Additional Notes

<!-- 추가 사항이 있을 경우, Todo list를 작성해주세요. -->

- [ ] (없음)

rak517 committed:
:recycle: refactor: 댓글 수정 시 변경사항이 없으면 submit 불가능하게 변경

rak517 committed:
:recycle: refactor: 댓글 리스트 영역 스크롤 부여로 모달 세로 길이 추가되지 않게 수정

ToKyun02 commented:
2개의 개선사항 적용하신 거 확인했습니다!

다음 페이지 페칭 상태와 최초 로딩 상태에 대한 로딩 구현도 해당 브랜치에서 추가 작업 해주시면 좋을 것 같습니다!

cksrlcks commented on src/components/dashboard/CommentSection.tsx:
```diff
@@ -109,29 +113,30 @@ export default function CommentSection({ cardId, columnId, dashboardId }: Commen
+                    <span className='text-md font-semibold text-gray-70'>{comment.author.nickname}</span>
+                    <span className='text-xs text-gray-40'>{formatDate(comment.createdAt)}</span>
+                  </div>
+                  <span className='text-md text-gray-70'>{comment.content}</span>
```
요기에 break-all 하나만 넣어주세욥

rak517 committed:
:recycle: refactor: 로딩 상태 스켈레톤 UI 추가 및 미세 UI 조정

ToKyun02 commented:
https://github.com/user-attachments/assets/eded64c8-985f-4c7f-a052-e49081ed6630

영상에 보이시는것처럼 코멘트 리스트 영역이 고정된 Height값이 아니라서 데이터가 없을 때 밀림 현상이 있습니다.(CLS)

이 부분만 고치시면 더 이상 개선할 건 없을 것 같습니다!

rak517 committed:
:recycle: refactor: 고정된 height 값으로 댓글 리스트 밀리는 현상 개선

rak517 committed:
:recycle: refactor: 댓글 리스트 로딩 개선

- 첫 로딩 시 스켈레톤 UI -> 텍스트 메시지로 변경
- isFetchingNextPage일 때만 스켈레톤 UI 렌더링
- 댓글 없을 때 화면 처리

cksrlcks reviewed (approved):
고생하셨습니다. !
보기 너무 좋아졌습니다.
